pkgs/applications/editors/emacs/sources.nix

```diff
@@ -34,55 +35,88 @@ let
     platforms = lib.platforms.all;
     mainProgram = "emacs";
   };
+
+  macportMetaFor = version: rev: {
+    homepage = "https://bitbucket.org/mituharu/emacs-mac/";
+    description = (mainlineMetaFor version).description + " - macport variant";
+    longDescription = (mainlineMetaFor version).longDescription + ''
+
+      This release is built from Mitsuharu Yamamoto's patched source code
+      tailored for MacOS X.
+    '';
+    changelog = "https://bitbucket.org/mituharu/emacs-mac/raw/${rev}/NEWS-mac";
+    inherit (mainlineMetaFor version) license maintainers mainProgram;
+    platforms = lib.platforms.darwin;
+  };
 in
 {
-  emacs28 = import ./generic.nix {
+  emacs28 = let
     pname = "emacs";
     version = "28.2";
     variant = "mainline";
+    rev = version;
+    hash = "sha256-4oSLcUDR0MOEt53QOiZSVU8kPJ67GwugmBxdX3F15Ag=";
+  in import ./generic.nix {
+    inherit pname version variant;
+
     src = fetchFromSavannah {
       repo = "emacs";
-      rev = "28.2";
-      hash = "sha256-4oSLcUDR0MOEt53QOiZSVU8kPJ67GwugmBxdX3F15Ag=";
+      inherit rev hash;
     };
 
-    meta = mainlineMeta;
+    meta = mainlineMetaFor version;
   };
 
-  emacs29 = import ./generic.nix {
+  emacs29 = let
     pname = "emacs";
-    version = "29.1-rc1";
+    version = "29.1";
     variant = "mainline";
+    rev = version;
+    hash = "sha256-3HDCwtOKvkXwSULf3W7YgTz4GV8zvYnh2RrL28qzGKg=";
+  in import ./generic.nix {
+    inherit pname version variant;
+
     src = fetchFromSavannah {
       repo = "emacs";
-      rev = "29.1-rc1";
-      hash = "sha256-p0lBSKsHrFwYTqO5UVIF/PgiqwdhYQE4oUVcPtd+gsU=";
+      inherit rev hash;
     };
 
-    meta = mainlineMeta;
+    meta = mainlineMetaFor version;
   };
 
-  emacs-macport = import ./generic.nix {
+  emacs28-macport = let
     pname = "emacs-mac";
     version = "28.2";
     variant = "macport";
+    rev = "emacs-28.2-mac-9.1";
+    hash = "sha256-Ne2jQ2nVLNiQmnkkOXVc5AkLVkTpm8pFC7VNY2gQjPE=";
+  in import ./generic.nix {
+    inherit pname version variant;
+
```

Comment:
What's up with this weird let here? Just put the values in directly.

Reply:
I'm figuring out how to automate the acquisition and generation of these values.
And concentrating them on this `let` makes it easy.

Reply:
Do that after you've figured it out. Until then, this code is to remain legible by humans, not bots.

Reply:
The code is still very legible by humans.
But I will not pursue this now, I am tired of bikeshedding these days.